core/web/job_runs_controller.go

```diff
@@ -54,17 +56,42 @@ func (jrc *JobRunsController) Create(c *gin.Context) {
 		jsonAPIError(c, http.StatusNotFound, errors.New("Job not found"))
 	} else if err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else if !j.WebAuthorized() {
-		jsonAPIError(c, http.StatusForbidden, errors.New("Job not available on web API, recreate with web initiator"))
+	} else if initiator, err := getAuthenticatedInitiator(c, j); err != nil {
+		jsonAPIError(c, http.StatusForbidden, err)
 	} else if data, err := getRunData(c); err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else if jr, err := services.ExecuteJob(j, j.InitiatorsFor(models.InitiatorWeb)[0], models.RunResult{Data: data}, nil, jrc.App.GetStore()); err != nil {
+	} else if jr, err := services.ExecuteJob(j, *initiator, models.RunResult{Data: data}, nil, jrc.App.GetStore()); err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 	} else {
 		jsonAPIResponse(c, presenters.JobRun{JobRun: *jr}, "job run")
 	}
 }
 
+// getInitiator returns the Job Spec's initiator for the given web context.
+func getAuthenticatedInitiator(c *gin.Context, js models.JobSpec) (*models.Initiator, error) {
+	if _, ok := authenticatedUser(c); ok {
+		webInitiators := js.InitiatorsFor(models.InitiatorWeb)
+		if len(webInitiators) == 0 {
+			return nil, errors.New("Job not available on web API, recreate with initiator type 'InitiatorWeb'")
+		}
+		return &webInitiators[0], nil
```

Comment:
Is there an actual case for this right now? If not I'd be hesitant to leave this case open given it's a security concern.

Reply:
I'm unsure.  In its defense, it preserves earlier functionality.  In order to trigger a Job Run via this, the Job Spec has to list 'InitiatorWeb' as one of its initiator types. i.e. its opt-in.
@se3000 what do you think?

Reply:
Web initiators are used by the Operator UI and the CLI, we use them pretty frequently.

Reply:
We should keep that in though, definitely helpful functionality.